### PR TITLE
Improvements and outlining feature

### PR DESCRIPTION
**Pull request by RocketRider**
I really like your library and I want to use it in a big project.
But there are some features I need there.

To use the library I needed to fix the manifest because of a know gradle issue.
I also added the outlining feature, I'm not completely satisfied with it, because on a white background it looks kind of strange because you only see the black outlinings.

Two more features I would like to add are to set the size of the items and to set a number of items per row.

### CODE REVIEW

**Comment by nwalters512** [comment]
Thanks for the contributions! In the future, if you could split your pulls into smaller chunks, it would be easier for me to review and merge.

I'll read through your work and leave comments where appropriate.

File: spectrum/src/main/res/values/attrs.xml [hunk 1]
```diff
@@ -1,14 +1,17 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
     <attr name="spectrum_colors" format="reference" />
+    <attr name="spectrum_strokeWidth" format="dimension" />
```
Comment (nwalters512) [inline spectrum/src/main/res/values/attrs.xml]: I'd consider renaming this `spectrum_borderWidth` or `spectrum_borderStrokeWidth`. I think it's important that it be self-evident that this applies to the border around the color.

File: spectrum/src/main/java/com/thebluealliance/spectrum/internal/SpectrumPreferenceDialogFragmentCompat.java [hunk 2]
```diff
@@ -14,7 +14,9 @@
 public class SpectrumPreferenceDialogFragmentCompat extends PreferenceDialogFragmentCompat {
 
     private SpectrumPalette mColorPalette;
-    private @ColorInt int mCurrentValue;
+    @ColorInt
```
Comment (nwalters512) [inline spectrum/src/main/java/com/thebluealliance/spectrum/internal/SpectrumPreferenceDialogFragmentCompat.java]: My convention is to leave single annotations on the same line as the variable they apply to.

File: spectrum/src/main/java/com/thebluealliance/spectrum/internal/SpectrumPreferenceDialogFragmentCompat.java [hunk 3]
```diff
@@ -14,7 +14,9 @@
...
+    private
```
Comment (nwalters512) [inline spectrum/src/main/java/com/thebluealliance/spectrum/internal/SpectrumPreferenceDialogFragmentCompat.java]: access modifier keywords should be on the same line as their declared variable

File: spectrum/src/main/java/com/thebluealliance/spectrum/internal/ColorItem.java [hunk 4]
```diff
@@ -56,16 +57,34 @@ public ColorItem(Context context, AttributeSet attrs) {
...
+    private void init() {
+        updateDrawables();
 
         mEventBus.register(this);
         setOnClickListener(this);
 
         LayoutInflater.from(getContext()).inflate(R.layout.color_item, this, true);
         mItemCheckmark = (ImageView) findViewById(R.id.selected_checkmark);
-        mItemCheckmark.setImageResource(isDarkBackground() ? R.drawable.ic_check_white_24dp : R.drawable.ic_check_black_24dp);
+        mItemCheckmark.setColorFilter(ColorUtil.isColorDark(mColor) ? Color.WHITE : Color.BLACK);
```
Comment (nwalters512) [inline spectrum/src/main/java/com/thebluealliance/spectrum/internal/ColorItem.java:76]: Nice! Better than using two different drawables.

**Comment by nwalters512** [comment]
Really nice work! I just had a few minor comments above.

I think we may want to differentiate between "should have a border" light and "should use a dark checkmark" light. There's a range of colors where you can visually distinguish them from the background, but it still looks better that they have a dark checkmark. With a range of all shades of gray, I found that bumping the brightness threshold to 240 or so will only start applying dark borders when it becomes hard to differentiate the colors from the background. It could maybe even go a little higher:

Imgur

Better yet, make that threshold (or both of them, if we split thresholds for checkmarks and borders) configurable by the user.

That may all be something for another PR.

**Comment by RocketRider** [comment]
Thank you for your feedback, yes I know I should have splitted the PR. Sry about that. I'm not used to PRs ^^
How will we proceed with these changes? Can you merge them somehow? Or should I create smaller PRs?

**Comment by nwalters512** [comment]
I'm happy merging this once we finish review, don't worry about it this time.

Do you have any thoughts on my comments?

File: spectrum/src/main/res/values/attrs.xml (hunk 1 above)
Comment (RocketRider) [inline spectrum/src/main/res/values/attrs.xml]: That is a good improvement, I was unsure about the name my self. 
borderStrokeWidth sounds better

File: spectrum/src/main/java/com/thebluealliance/spectrum/internal/SpectrumPreferenceDialogFragmentCompat.java (hunk 3 above)
Comment (RocketRider) [inline spectrum/src/main/java/com/thebluealliance/spectrum/internal/SpectrumPreferenceDialogFragmentCompat.java]: Yes you are right, autoformatter did that

File: spectrum/src/main/res/values/attrs.xml (hunk 1 above)
Comment (nwalters512) [inline spectrum/src/main/res/values/attrs.xml]: The more I think about it, I like `borderWidth` even better. The fact that it's a stroke isn't really relevant.

**Comment by nwalters512** [comment]
@RocketRider do you still want to finish up this PR?

**Comment by RocketRider** [comment]
Yes I will take care of the changes when I have the time...

**Comment by RocketRider** [comment]
So I fixed the code.

**Comment by nwalters512** [comment]
Thanks! I'll release a new version of the library today.

**Comment by RocketRider** [comment]
Thank you, that would be amazing ;)

**Pull request merged**